go-controller/pkg/ovn/ovn.go

```diff
@@ -301,61 +289,18 @@ func (oc *Controller) Run(wg *sync.WaitGroup, nodeName string) error {
 	// dependencies, and WatchNodes() depends on it
 	oc.WatchNamespaces()
 
+	// Services must be started before nodes for handling new node's service sync
+	if err := oc.StartServiceController(wg, true); err != nil {
+		return err
+	}
+
 	// WatchNodes must be started next because it creates the node switch
 	// which most other watches depend on.
 	// https://github.com/ovn-org/ovn-kubernetes/pull/859
 	oc.WatchNodes()
```

Comment:
this worries me, so let's say we deploy/upgrade a deployment with 100 nodes.
1x SyncServices()
1x SyncNodes() --> 100 x SyncServices()

Reply:
Yeah, is it worth me adding the logic that will only add affected services to the queue? Since our nbctl is not optimal it seems like it might be worth it at this time.

Reply:
After discussing with Antonio, let's hold off on this for now.